gflownet/envs/grid.py

```diff
@@ -62,15 +83,21 @@ def __init__(
 
     def get_action_space(self):
         """
-        Constructs list with all possible actions, including eos.
+        Constructs list with all possible actions, including eos. An action is
+        represented by a vector of length n_dim where each index d indicates to
+        increment to apply to dimension d of the hyper-grid.
         """
-        valid_steplens = np.arange(self.min_step_len, self.max_step_len + 1)
-        dims = [a for a in range(self.n_dim)]
         actions = []
-        for r in valid_steplens:
-            actions_r = [el for el in itertools.product(dims, repeat=r)]
-            actions += actions_r
-        actions += [(self.eos,)]
+        for incr in range(self.max_increment + 1):
```

Comment:
I don't think this works for `max_dim_per_action > 1`, if I understand correctly, for example for `n_dim = 2` and `max_dim_per_action = 2`, the expected result should be `[(1, 0), (0, 1), (1, 1)]`, and it's `[(1, 0), (0, 1)]`?

If I understood the intended behavior correctly I would recommend writing a test for that with some expected, predefined values.

Reply:
Thanks for the catch, my bad :( I have re-implemented this properly and added an explicit test for it.